File: src/Shared/Polyrific.Catapult.Shared.Dto/ExternalServiceType/ExternalServicePropertyDto.cs
// Copyright (c) Polyrific, Inc 2018. All rights reserved.

namespace Polyrific.Catapult.Shared.Dto.ExternalServiceType
{
    public class ExternalServicePropertyDto
    {
        /// <summary>
        /// Name of the external service property
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Description of the external service property
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Allowed values of the external service property
        /// </summary>
        public string[] AllowedValues { get; set; }

        /// <summary>
        /// Indicates whether the external service property is required
        /// </summary>
        public bool IsRequired { get; set; }

        /// <summary>
        /// Indicates whether the external service property is a secret property that needs to be obfuscated in UI
        /// </summary>
        public bool IsSecret { get; set; }
    }
}

File: src/Web/Polyrific.Catapult.Web/ClientApp/src/app/core/models/external-service/external-service-property-dto.ts
import { AdditionalLogicDto } from './additional-logic-dto';

export interface ExternalServicePropertyDto {
  name: string;
  description: string;
  allowedValues: string[];
  isRequired: boolean;
  isSecret: boolean;
  sequence: number;
  additionalLogic: AdditionalLogicDto;
}
